diff -Naur celery.orig/apps/beat.py celery/apps/beat.py
--- celery.orig/apps/beat.py	2017-01-11 09:58:24.547860000 +0100
+++ celery/apps/beat.py	2017-01-11 09:59:23.326613068 +0100
@@ -114,6 +114,7 @@
             logger.critical('beat raised exception %s: %r',
                             exc.__class__, exc,
                             exc_info=True)
+            raise
 
     def init_loader(self):
         # Run the worker init handler.
